src/targz_sizes.c

```diff
@@ -137,7 +159,11 @@ int main(int argc, char** argv) {
     inflateGetHeader(&gzip_stream, &gzip_header);
 
     // state machine
+    int file_blocks = 0;
     int skip_blocks = 0;
+    int read_blocks = 0;
+    unsigned char* long_filename = (unsigned char*) filename_buff;
```

Comment:
You're breaking the strict aliasing rule here. For explanation, see e.g. https://stackoverflow.com/questions/98650/what-is-the-strict-aliasing-rule. For this code, just declare the original `filename_buff` as an array of chars with an appropriate size, should be enough for this code.